### PR TITLE
Allow parentheses in single-line inheritance

### PR DESCRIPTION
**Pull request by gsamokovarov**
`class Point < Struct.new(:x, :y); end` was getting auto-corrected to
`class Point < Struct.new :x, :y; end` which is invalid Ruby code. Allow
the parentheses in this case for the following configurations:

```ruby
Style/MethodCallWithArgsParentheses:
  EnforcedStyle: omit_parentheses
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Bug fixes
 
+* [#8346](https://github.com/rubocop-hq/rubocop/issues/8346): Allow parentheses in single-line inheritance with `Style/MethodCallWithArgsParentheses` `EnforcedStyle: omit_parentheses` to fix invalid Ruby auto-correction. ([@gsamokovarov][])
```
Comment (koic) [inline CHANGELOG.md:7]:
```suggestion
* [#8346](https://github.com/rubocop-hq/rubocop/pull/8346): Allow parentheses in single-line inheritance with `Style/MethodCallWithArgsParentheses` `EnforcedStyle: omit_parentheses` to fix invalid Ruby auto-correction. ([@gsamokovarov][])
```

File: lib/rubocop/cop/style/method_call_with_args_parentheses/omit_parentheses.rb [hunk 2]
```diff
@@ -142,7 +147,7 @@ def ternary_if?(node)
           end
 
           def logical_operator?(node)
-            (node.and_type? || node.or_type?) && node.logical_operator?
+            node && (node.and_type? || node.or_type?) && node.logical_operator?
```
Comment (koic) [inline lib/rubocop/cop/style/method_call_with_args_parentheses/omit_parentheses.rb:150]: Is this change necessary?

Comment (gsamokovarov) [reply lib/rubocop/cop/style/method_call_with_args_parentheses/omit_parentheses.rb:150]: It's not, I'll go back and remove it.

**Comment by gsamokovarov** [comment]
Thanks for the review, @koic, can you check the changes now?
